src/components/data/data-utils.js

```diff
@@ -585,6 +585,66 @@ export const getAttributeType = attributeValue => {
   return { type, isList }
 }
 
+export const AttributeTypeInput = ({ label: labelText = 'Type', value, onChange, entityTypes = [], defaultReferenceEntityType = null, showJsonTypeOption = false }) => {
+  const { type, entityType: referenceEntityType } = value
+
+  const typeOptions = [
+    { type: 'string' },
+    { type: 'reference', tooltip: 'A link to another entity' },
```

Comment:
TOL: have we settled on what we want to actually call entities for user-facing text? (out of scope for this PR)

Reply:
Yeah, I think the design attached to [SU-99](https://broadworkbench.atlassian.net/browse/SU-99) updated the user facing labels. "Row" would probably be most consistent with the rest of the data table UI.

Reply:
Updated in https://github.com/DataBiosphere/terra-ui/pull/3307/commits/7c047d4d3c0851a00c2f40abc38a0ac0c0ecb036.